test/e2e/stargate_apis_test.go

```diff
@@ -209,18 +225,20 @@ func readDocument(t *testing.T, restClient *resty.Client, k8sContextIdx int, tok
 		SetHeader("Content-Type", "application/json").
 		SetHeader("X-Cassandra-Token", token).
 		Get(url)
-	assert.NoError(t, err, "Failed to retrieve Stargate document")
+	require.NoError(t, err, "Failed to retrieve Stargate document")
 	var genericJson map[string]interface{}
 	err = json.Unmarshal(response.Body(), &genericJson)
-	assert.NoError(t, err, "Failed to decode Stargate response")
+	require.NoError(t, err, "Failed to decode Stargate response")
 	assert.Equal(t, documentId, genericJson["documentId"], "Expected JSON payload to contain a field documentId")
+	require.NotNil(t, genericJson["data"], "Expected JSON payload to contain a field data")
+	require.IsType(t, map[string]interface{}{}, genericJson["data"], "Expected field data to be map[string]interface{}")
 	assert.Equal(t, awesomeMovieDirector, genericJson["data"].(map[string]interface{})["Director"], "Expected JSON payload to contain a field data.Director")
 	assert.Equal(t, awesomeMovieName, genericJson["data"].(map[string]interface{})["Name"], "Expected JSON payload to contain a field data.Name")
 }
 
 func authenticate(t *testing.T, restClient *resty.Client, k8sContextIdx int, username, password string) string {
 	url := fmt.Sprintf("http://stargate.127.0.0.1.nip.io:3%v080/v1/auth", k8sContextIdx)
-	body := fmt.Sprintf("{\"username\": \"%s\", \"password\": \"%s\"}", username, password)
+	body := map[string]string{"username": username, "password": password}
 	request := restClient.NewRequest().
```

Comment:
Would it make sense and help to wrap the request in an `Eventually` call?

Reply:
The very first time we authenticate is actually done inside `DeployStargateIngresses` and that call is wrapped in an `Eventually` block. If that `Eventually` succeeds there should be no reason for this call here to fail.

Reply:
Also, this call here was failing previously because in `DeployStargateIngresses` we weren't doing a full authentication, we were just pinging the auth endpoint.